File: SDKs/amazonaws.com/events/2015-10-07/python/src/sdk/models/shared/runcommandparameters.py
from dataclasses import dataclass, field
from typing import List
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class RunCommandParameters:
    r"""RunCommandParameters
    This parameter contains the criteria (either InstanceIds or a tag) used to specify which EC2 instances are to be sent the command. 
    """
    
    run_command_targets: List[RunCommandTarget] = field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('RunCommandTargets') }})
    

File: SDKs/amazonaws.com/events/2015-10-07/typescript/src/sdk/models/shared/runcommandparameters.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { RunCommandTarget } from "./runcommandtarget";



// RunCommandParameters
/** 
 * This parameter contains the criteria (either InstanceIds or a tag) used to specify which EC2 instances are to be sent the command. 
**/
export class RunCommandParameters extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=RunCommandTargets", elemType: RunCommandTarget })
  runCommandTargets: RunCommandTarget[];
}
